test/integration/preflight/preflight_test.go

```diff
@@ -0,0 +1,120 @@
+package preflight
```

Comment:
>>Can't say I'm a huge fan of using golang for testing here, seems like it is going to be difficult to get external contributors to participate if the tests are written in golang.

>Why do you think so?

Most external contributors will not be familiar with golang, If they are already advanced Ansible users, then we can generally assume they are familiar with Python.

>>I'd much prefer that we move towards a framework like https://molecule.readthedocs.io

>Thanks for the link. I see how it could be useful for us in end-to-end testing of the many playbooks in openshift-ansible (installation, upgrades, one-off actions). IIUC it's a YAML-centric framework for a particular flow of testing playbooks? I see it can do check syntax, convergence, idempotence, verify final state.

The verify final state is the one that allows for arbitrary validations to be run.

>For testing the preflight checks, what we need is a way to run openshift-ansible someplaybook.yml and check the exit code and output. Each test playbook will setup a dynamic inventory composed of one or more containers, and then will include another playbook from the preflight checks, to ensure that the check work correctly under certain environments (e.g. system with missing subscriptions, or incompatible versions of packages).

I'll play around with it some, I'm pretty sure this use case is baked in, even if the docs are lacking in how to do it.

>We wanted something that could give us fast feedback to iterate on the checks. Thus, we use Docker images instead of VMs. Running the tests in parallel seemed desirable, and Go's testing let us do that without much hassle.

Molecule is able to use Docker as a backend as well. I'll have to play around with the parallelization, but I think we can accomplish something similar.